dbms/tests/queries/0_stateless/01055_compact_parts_granularity.sql

```diff
@@ -0,0 +1,21 @@
+drop table if exists mt_compact;
+
+-- Checks that granularity correctly computed from small parts.
+
+create table mt_compact(a Int, s String) engine = MergeTree order by a
+settings min_rows_for_wide_part = 1000,
+index_granularity = 14;
+
+system stop merges mt_compact;
+set max_block_size = 1;
+set min_insert_block_size_rows=1;
+insert into mt_compact select number, 'aaa' from numbers(100);
+
+select count() from system.parts where table = 'mt_compact' and database = currentDatabase() and active;
```

Comment:
i'm wondering - will system.columns and part_columns be able to show proper sizes for those parts.

Shouldn't those parts be marked in system.parts somehow?

Reply:
> i'm wondering - will system.columns and part_columns be able to show proper sizes for those parts.

No, per-column sizes wouldn't be counted for compact parts. Count compressed size of every column is almost impossible. Uncompressed size can be counted, but it's not very usefull, and it's not implemented for simplicity.

> Shouldn't those parts be marked in system.parts somehow?

`system.parts` table would have `part_type` column.

Reply:
That will definitely be reported as bug ("size on disk doesn't match the size in system.columns" etc.) :\ AFAIK a lot of users look in system.columns.

Reply:
Compact parts are intended to have only small ratio in size to total size of the table.